In [1]:
import numpy as np
import pandas as pd
import statistics
import matplotlib as mtpl
import matplotlib.pyplot
%matplotlib inline

Question 3: Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?

Mean Calculation Options:

In [2]:
#formula: mean = sum(x) / n
mean = (14 + 12 + 11 + 11 + 8) / 5
mean

11.2

In [3]:
#Using Python: mean = sum(df['age']) / len(df['age'])

names = ['Greg','Marcia','Peter', 'Jan','Bobby', 'Cindy','Oliver']
ages = [14, 12, 11, 11, 8]
         
mean = sum(ages) / len(ages)
mean


11.2

In [4]:
#Using Numpy:
mean = np.mean(ages)
mean


11.2

Median Calculation Options:

In [5]:
#Median
#formula: step 1. Order the numbers from smallest to largest.  step 2. (n+1) / 2

for age in ages:

    ages = [int(age) for age in ages]
    ages.sort()
    ages.sort()
    
#Below is my guess
#Question for Mike:
ages.sort()

for age in ages:

    median = ((age) + 1) / 2
    print(median)

4.5
6.0
6.0
6.5
7.5


In [6]:
#Using the Python Statistics library
median = statistics.median(ages)
median

11

In [7]:
#Using Numpy
np.median(ages)

11.0

Mode Calculation Options:

In [13]:
#Code:
#From the Statistics library use: statistics.mode(ages)

statistics.mode(ages)


11

Variance Calculation Options:

In [8]:
#v=sum(x-mean) ** 2 / (n-1)
ls =[]
for age in ages:

    ls.append((age - 11)**2)

variance = sum(ls) / ((len(ages)) - 1)
variance

4.75

Standard Deviation Calculattion: 

In [9]:
#Standard Deviation: s = v ** 0.5
standard_deviation = variance ** 0.5
standard_deviation

2.179449471770337

In [10]:
#Numpy Standard Deviation method:
np.std((ages), ddof = 1)

2.16794833886788

Standard Error Calculation:

In [11]:
#se = s / (n ** 0.5)
#Note: s represents the sample
standard_error = standard_deviation / (len(ages) ** 0.5)
standard_error

0.9746794344808964

In [12]:
#np.std(df['age'] ,ddof=1) / np.sqrt(len(df['age']))
np.std(ages, ddof=1) / np.sqrt(len(ages))

0.9695359714832658

Question 3 Response: 
Observations:

Changes in Central Tendency Values: 
Mean: The median has increased by 0.2, and is now 11.2.
Median: The meidan remains 11.0.
Mode: There is now a mode.  11 is the mode. 

Changes in Variance Values:
Variance: The variance has increased by 0.25, and is not 4.75.

Standard Deviation: The standard deviation without a ddof = 1 increased by 0.4473986642014598. (Is this answer wrong?  Should the ddof always be addes.  This result seems problematic.
                    
The standard deviation with ddof = 1 decreased by 0.06811963863190984.  This is due to the fact that now that two of the five numbers appears twice, the values are not scatterd as broadly as they were when no number appeared twice.  Therefore, there is less deviation between values. 

Standard Error: The standard error without ddof = 1 has increased by 0.20008276523941315.

The standard error with ddof = 1 has decreased by 0.030464028516734243.

Question 3, Response

Conclusions:

Mean: Because the sum of (ages) only increased by 1. the mean was not greatly impacted.

Median: The median was not impacted at all by the addition of a value of 11, due to the fact that a value of 11 was already the middle value in the data set.  

Mode: Because the original data set did not contain a repeated value, the replacement of 10 with 11 created a data set which contained the value of 11 in two positions or elements.  This change, therefore, allowed for a mode, and had a significant impact on the mode, allowing it to became an applicable measure of central tendency for the data set in question.

Variance: The variance has increased by 0.25, because now, we have a large gap between two of our values (i.e. a larger gap in the spread of the value, becaue ther is a greater distance between 8 and 11 than their is between 8 and 10.

Standard Deviation:  Consequently, as the standard deviation is simply a restatement of the variance in a different units of measurement between the data points (namely a unit that is not squared) the standard deviation may also be observed to have increased.  

However, if ddof = 1 is added to the calculation of standard deviation, the resulting value is greater than the original value.

Standard Error:  The pattern can seen in standard error when comparing values for quetion 1 and question 2.  The standard error (which is useful in that, when given multiple sets of data, it finds the mean of the means and based on this measurement provides an average standard deviation) presents the same pattern.  The value decreases, unless the ddof = 1 is added to the calculation, in which case the value increases. 



Questions:  

1. Should ddof = 1 always be part of any calcultion of standard deviation and consequently of standard error?  

2. Does the ddof change depending on how many sample measurements (i.e. data sets) a data scientist is evaluating?
